### PR TITLE
Support bind schemas and OSB fields in bundles

### PR DESCRIPTION
**Pull request by polskikiel**
**Description**

Changes proposed in this pull request:

- HelmBroker support for `ServicePlan` field `Free`.
(provided by plan/meta.yaml)
- HelmBroker support for `ServiceClass` fields `BindingRetrievable`, `Requires`, `PlanUpdatable`. (provided by meta.yaml)
- HelmBroker support for bind-schemas

**Related issue(s)**

### CODE REVIEW

**Comment by piotrmiskiewicz** [comment]
Please update the documentation

File: docs/service-brokers/docs/012-configuration-helm-broker-bundles.md [hunk 1]
```diff
@@ -47,6 +47,9 @@ The `meta.yaml` file is mandatory as it contains information about the bundle. S
 |   **documentationURL**  |   false  |                                  The **documentationURL** field described in the [Open Service Broker API][service-metadata].                                    |
 |      **supportURL**     |   false  |                                     The **supportURL** field described in the [Open Service Broker API][service-metadata].                                       |
 |       **imageURL**      |   false  |     The **imageURL** field described in the [Open Service Broker API][service-metadata]. You must provide the image as an SVG.          |
+| **bindingRetrievable**  |   false  |     The **bindingRetrievable** field represents whether fetching a service binding via a GET on the binding resource's endpoint is supported for all plans. The default value is `false`.         |
+|   **planUpdatable**     |   false  |     The **planUpdatable** field PlanUpdatable represents whether instances of this service may be updated to a different plan. The default value is `false`         |
```
Comment (klaudiagrz) [inline docs/service-brokers/docs/012-configuration-helm-broker-bundles.md]:
```suggestion
|   **planUpdatable**     |   false  |     The **planUpdatable** field specifies whether instances of this service can be updated to a different plan. The default value is `false`         |
```

File: docs/service-brokers/docs/012-configuration-helm-broker-bundles.md [hunk 2]
```diff
@@ -47,6 +47,9 @@ The `meta.yaml` file is mandatory as it contains information about the bundle. S
...
+|       **requires**      |   false  |     The **requires** field defines a list of permissions the user must give instances of this service.          |
```
Comment (klaudiagrz) [inline docs/service-brokers/docs/012-configuration-helm-broker-bundles.md]:
```suggestion
|       **requires**      |   false  |     The **requires** field defines a list of permissions the user must grant to the instances of this service.          |
```

File: docs/service-brokers/docs/012-configuration-helm-broker-bundles.md [hunk 3]
```diff
@@ -72,6 +75,7 @@ The `meta.yaml` file contains information about a bundle plan. Set the following
 | **description** |   true   | The plan description. It has the same restrictions as defined in the [Open Service Broker API][plan-objects]. |
 | **displayName** |   true   | The plan display name. It has the same restrictions as defined in the [Open Service Broker API][plan-objects]. |
 |  **bindable**   |   false  | The plan bindable attribute. It has the same restrictions as defined in the [Open Service Broker API][plan-objects].    |
+|     **free**    |   false  | The plan free attribute.  The default value is `false`    |
```
Comment (klaudiagrz) [inline docs/service-brokers/docs/012-configuration-helm-broker-bundles.md]:
```suggestion
|     **free**    |   false  | The attribute which specifies whether the instance is free or not. The default value is `false`.    |
```

File: docs/service-brokers/docs/012-configuration-helm-broker-bundles.md [hunk 4]
```diff
@@ -47,6 +47,9 @@ The `meta.yaml` file is mandatory as it contains information about the bundle. S
 |   **documentationURL**  |   false  |                                  The **documentationURL** field described in the [Open Service Broker API][service-metadata].                                    |
 |      **supportURL**     |   false  |                                     The **supportURL** field described in the [Open Service Broker API][service-metadata].                                       |
 |       **imageURL**      |   false  |     The **imageURL** field described in the [Open Service Broker API][service-metadata]. You must provide the image as an SVG.          |
+| **bindingRetrievable**  |   false  |     The **bindingRetrievable** field specifies whether fetching a ServiceBinding using a `GET` call on the resource's endpoint is supported for all plans. The default value is `false`.         |
+|   **planUpdatable**     |   false  |     The **planUpdatable** field specifies whether instances of this service can be updated to a different plan. The default value is `false`         |
+|       **requires**      |   false  |     The **requires** field defines a list of permissions the user must grant to the instances of this service.          |
```
Comment (PK85) [inline docs/service-brokers/docs/012-configuration-helm-broker-bundles.md]: Describe bind-instance-schema and update-instance-schema. I see that also "labels" are not documented, whats about metadata, labels are a subset of metadata.

File: components/helm-broker/internal/broker/catalog.go [hunk 5]
```diff
@@ -66,13 +67,16 @@ func (f *bundleToServiceConverter) Convert(b *internal.Bundle) (osb.Service, err
...
-		Description: b.Description,
-		Bindable:    b.Bindable,
-		Plans:       sPlans,
-		Metadata:    meta.ToMap(),
-		Tags:        sTags,
+		ID:                 string(b.ID),
+		Name:               string(b.Name),
+		Description:        b.Description,
+		Bindable:           b.Bindable,
+		BindingRetrievable: b.BindingRetrievable,
```
Comment (PK85) [inline components/helm-broker/internal/broker/catalog.go]: OSB API has bindings_retrievable.

File: docs/service-brokers/docs/012-configuration-helm-broker-bundles.md [hunk 6]
```diff
@@ -47,6 +47,9 @@ The `meta.yaml` file is mandatory as it contains information about the bundle. S
 |   **documentationURL**  |   false  |                                  The **documentationURL** field described in the [Open Service Broker API][service-metadata].                                    |
 |      **supportURL**     |   false  |                                     The **supportURL** field described in the [Open Service Broker API][service-metadata].                                       |
 |       **imageURL**      |   false  |     The **imageURL** field described in the [Open Service Broker API][service-metadata]. You must provide the image as an SVG.          |
+| **bindingRetrievable**  |   false  |     The **bindingRetrievable** field specifies whether fetching a ServiceBinding using a `GET` call on the resource's endpoint is supported for all plans. The default value is `false`.         |
```
Comment (PK85) [inline docs/service-brokers/docs/012-configuration-helm-broker-bundles.md]: please point me to docs that default value is false.

File: docs/service-brokers/docs/012-configuration-helm-broker-bundles.md (hunk 4 above)
Comment (polskikiel) [reply docs/service-brokers/docs/012-configuration-helm-broker-bundles.md]: Described bind-instance-schema and update-instance-schema. Added description of labels field. Metadata discussed.

File: docs/service-brokers/docs/012-configuration-helm-broker-bundles.md (hunk 6 above)
Comment (polskikiel) [reply docs/service-brokers/docs/012-configuration-helm-broker-bundles.md]: I can't find a doc with this. But if we don't specify `bindingRetrievable` in `meta.yaml`, the default value for it is false in the ServiceClass.

Comment (piotrmiskiewicz) [reply docs/service-brokers/docs/012-configuration-helm-broker-bundles.md]: Service-catalog does not contain any logic with the default value, so it is `false`

File: components/helm-broker/Gopkg.toml [hunk 7]
```diff
@@ -36,8 +36,8 @@ required= [
   revision = "bdcc60b419d136a85cdf2e7cbcac34b3f1cd6e57"
 
 [[constraint]]
-  name = "google.golang.org/grpc"
-  branch = "v1.7.x"
```
Comment (piotrmiskiewicz) [inline components/helm-broker/Gopkg.toml]: why did you remove the `google.golang.org/grpc`entry?

File: docs/service-brokers/docs/012-configuration-helm-broker-bundles.md [hunk 8]
```diff
@@ -91,6 +98,18 @@ The `create-instance-schema.json` file contains a schema used to define the para
...
 
+#### The bind-instance-schema.json file
+
+The `bind-instance-schema.json` file contains a schema used to define the parameters. Each input parameter is expressed as a property within a JSON object.
+This file is not required.
+For more information about the content of the `bind-instance-schema.json` file, see the [Input parameters](https://github.com/openservicebrokerapi/servicebroker/blob/v2.14/spec.md#input-parameters-schema-object) specification.
+
+#### The update-instance-schema.json file
+
+The `update-instance-schema.json` file contains a schema used to define the parameters. Each input parameter is expressed as a property within a JSON object.
```
Comment (piotrmiskiewicz) [inline docs/service-brokers/docs/012-configuration-helm-broker-bundles.md]: Ther eis no any comment it is about an update (or the binding in the previous) operation

File: components/helm-broker/Gopkg.toml (hunk 7 above)
Comment (polskikiel) [reply components/helm-broker/Gopkg.toml]: discussed

File: docs/service-brokers/docs/012-configuration-helm-broker-bundles.md (hunk 8 above)
Comment (polskikiel) [reply docs/service-brokers/docs/012-configuration-helm-broker-bundles.md]: I developed each sentence to mention an operation to which it applies.
